types/googlemaps/reference/info-window.d.ts

```diff
@@ -1,11 +1,11 @@
 declare namespace google.maps {
-    interface InfoWindowHandlerMap {
+    interface InfoWindowHandlerMap<T extends InfoWindow> {
         /**
          * This event is fired when the close button was clicked.
          * @see {@link https://developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindow.closeclick Maps JavaScript API}
          * @see {@link InfoWindow#close}
          */
-        closeclick: [];
+        closeclick: (this: T) => void;
```

Comment:
I'm not sure I understand how this property could go from being an empty tuple to being a function, but maybe it makes sense in the context of the library?  From the doc link, it seems like the old type might just have been incorrect?

Reply:
Earlier I used `MVCEventHandler` to expand a tuple in a function.

```ts
type MVCEventHandler<T extends MVCObject, A extends any[]> = (this: T, ...args: A) => void;
```

```ts
addListener<N extends keyof InfoWindowHandlerMap>(
    eventName: N,
    handler: MVCEventHandler<this, InfoWindowHandlerMap[N]>,
): MapsEventListener;
```

For now, I just expanded this type to eliminate the empty tuple problem.

```ts
addListener<N extends keyof InfoWindowHandlerMap<this>>(
    eventName: N,
    handler: InfoWindowHandlerMap<this>[N],
): MapsEventListener;
```

I've added tests that check the correctness of types in the handlers. In version <3.5 they are evaluated in `any`. So it looks like the minimum version is 3.5. Or we will have to use overloads, which I really would not like.

Reply:
I see - the type used to map handler names to the types of their arguments, but now it maps to their function types.  I thought this was directly modelling the underlying library.

Reply:
I guess I would intuitively have expected overloads, but this seems okay.